test/extensions/filters/http/oauth2/filter_test.cc

```diff
@@ -574,10 +576,30 @@ TEST_F(OAuth2Test, OAuthOptionsRequestAndContinue) {
       {Http::Headers::get().Path.get(), "/anypath"},
       {Http::Headers::get().Method.get(), Http::Headers::get().MethodValues.Options},
   };
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+}
+
+/**
+ * Scenario: Authorization header is one of the headers in pass_through_matcher
+ *
+ * Expected behavior: Passthrough unaltered request (especially Authorization header)
```

Comment:
This test makes it sound like the behavior changes when the authorization header is used as one of the matchers, but I don't think that's the case? Can we just include a check to the existing pass through test that shows that the authorization header is left untouched?